Sprint 19a

fury-sound opened the pull request:
верстка основных экранов с выводом моковых данных, для навигации и связи использованы биндинги, частично реализован MVVM; темная тема внедрена, вывод пользовательского соглашения через WebView; комментарии оставлены для дальнейшего использоования (будут удалены позже)
Текущая ветка c с работающим кодом - sprint19a

fury-sound committed:
sprint19 - файлы добавлены в ветку 19, внесены изменения по задаче. Комментарии в коде оставлены для дальнейшего использования.

fury-sound committed:
sprint 19 - исправление структуры проекта после сбоя гита

fury-sound committed:
sprint 19 - исправления структуры проекта после сбоя гита

fury-sound committed:
sprint 19 - корректировка кода под 16 версию iOS

fury-sound committed:
sprint 19 - поправки UI

NaR0RoW commented on TravelSchedule/TravelScheduleApp.swift:
```diff
@@ -10,14 +10,15 @@ import SwiftUI
 @main
 struct TravelScheduleApp: App {
 //    @State var routeDirection = RouteDirection()
-    @StateObject var routeDirection = RouteDirection()
+//    @StateObject var routeDirection = RouteDirection()
```
Закомментированный код лучше удалять перед review

NaR0RoW commented on TravelSchedule/Views/CarrierSearch.swift:
```diff
@@ -52,23 +52,39 @@ struct CarrierSearch: View {
                             }
                             .listRowSeparator(.hidden)
                         }
+                        .ignoresSafeArea(.all)
```
`.all` можно опустить, т.к. это значение по умолчанию:
`.ignoresSafeArea()`

NaR0RoW commented on TravelSchedule/Views/CarrierSearch.swift:
```diff
@@ -52,23 +52,39 @@ struct CarrierSearch: View {
                     VStack {
                         List {
```
Кажется List можно убрать

NaR0RoW commented on TravelSchedule/Views/LocationSelection.swift:
```diff
@@ -20,8 +20,6 @@ struct LocationSelection: View {
     var body: some View {
-            //        Text("")
-            //        CityListTable(path: $path, routeData: routeData)
         if noInternetError == true {
```
`== true` можно опустить:
`if noInternetError {`

NaR0RoW replied:
Везде, где это возможно, лучше использовать `opacity` вместо `if` для сокрытия `View`. [Вот хороший пример с ссылками на `WWDC`](https://stackoverflow.com/a/57420479)

NaR0RoW commented on TravelSchedule/Views/LocationSelection.swift:
```diff
@@ -36,19 +34,29 @@ struct LocationSelection: View {
                     .font(.system(size: 17, weight: .regular))
-//                    .navigationTitle(headerText)
-            }.toolbarVisibility(.hidden, for: .tabBar)
+//                    .transition(.asymmetric(
+//                        insertion: AnyTransition.scale(scale: 0.1, anchor: .leading).combined(with: .opacity),
+//                        removal: .move(edge: .trailing)))
+            }
```
Форматирование кода немного поехало, можно выделить это место и воспользоваться автоматическим форматированием в Xcode `control + I`

NaR0RoW commented on TravelSchedule/Views/LocationSelection.swift:
```diff
@@ -106,23 +114,24 @@ struct CityListTable: View {
                     }
                     .listRowSeparator(.hidden)
-                }.listStyle(.inset)
+                    .padding([.top, .bottom], 10)
```
`[.top, .bottom]` - это тоже самое, что и `.vertical`:
`.padding(.vertical, 10)`

fury-sound committed:
sprint 19 - небольшие правки UI

fury-sound committed:
sprint 19 - исправления критических замечаний ревьюера: дизайн экрана пользовательского соглашения исправлен, темная тема реализована; цвет заднего фона на всех экранах задается под дизайн; отступы на экране настроек маршрута откорректированы; заголовок "Информация о перевозчике" добавлен на экран карточки перевозчика;  на экране выбора станции удален заголовка с именем выбранного города
добавлено еще немного анимации (кнопка Утонить время); мелкие исправления

fury-sound committed:
sprint 19 - корректировки отображения цветовой темы

fury-sound commented:
подправил критические замечания, исправления по дополнитльным замечаниям внесу позже, в ходе работ по 20 спринту

NaR0RoW commented on TravelSchedule/ViewModel/RouteCarrierData.swift:
```diff
@@ -9,9 +9,9 @@ import Foundation
-    let carrierRzd: CarrierDetails = CarrierDetails(id: .rzd, name: "OАO РЖД", imageNameSmall: "rzd")
-    let carrierFgk: CarrierDetails = CarrierDetails(id: .fgk, name: "OАO ФГК", imageNameSmall: "fgk")
-    let carrierUral: CarrierDetails = CarrierDetails(id: .uralLogistika, name: "OАO Урал логистика", imageNameSmall: "urallogistic")
+    let carrierRzd: CarrierDetails = CarrierDetails(id: UUID(), name: .rzd, nameLong: "OАO РЖД", imageNameSmall: "rzd")
```
`UUID()` можно не передавать, а сразу внутри структуры создавать:
``` swift  
var id: UUID = UUID()
```

NaR0RoW commented on TravelSchedule/Models/Carrier.swift:
```diff
@@ -10,12 +10,13 @@ import Foundation
 struct CarrierDetails: Identifiable, Hashable {
-    var id: CarrierID
-    var name: String
+    var id: UUID
+    var name: CarrierID
+    var nameLong: String
```
Лучше иммутабельные структуры делать, то есть использовать поля - константы, а не переменные. Сами поля можно менять с помощью методов, которые будут возвращать новую структуру с измененными полями

NaR0RoW commented on TravelSchedule/Views/userAgreementView.swift:
```diff
@@ -9,53 +9,213 @@ import SwiftUI
+    let blackBGColor: String = "1A1B22"
+        //    @Environment(\.colorScheme) var colorScheme
+
+    class Coordinator: NSObject, WKNavigationDelegate {
```
Стоит пометить как `final`. Если класс не наследуется, то лучше пометить его как `final` для ускорения работы компилятора

NaR0RoW commented on TravelSchedule/Views/userAgreementView.swift:
```diff
@@ -9,53 +9,213 @@ import SwiftUI
 struct WebView: UIViewRepresentable {
-    let urlString: String
+    @Binding var isDarkMode: Bool
+    let blackBGColor: String = "1A1B22"
```
Для достижения [инкапсуляции](https://habr.com/ru/articles/87205/) лучше пометить это поле `private`

NaR0RoW commented on TravelSchedule/Views/userAgreementView.swift:
```diff
@@ -9,53 +9,213 @@ import SwiftUI
+        webView.isOpaque = false
+        webView.backgroundColor = .clear
+        webView.scrollView.backgroundColor = .clear
+        let url = URL(string: "https://www.yandex.ru/legal/practicum_offer/")!
```
Давай не будем `force` операции использовать

NaR0RoW commented on TravelSchedule/Views/userAgreementView.swift:
```diff
@@ -9,53 +9,213 @@ import SwiftUI
     }
 }
 
 struct userAgreementView: View {
```
Название структур должно с большой буквы начинаться

NaR0RoW commented on TravelSchedule/Views/userAgreementView.swift:
```diff
@@ -9,53 +9,213 @@ import SwiftUI
-//    let urlString: String = "https://yandex.ru/legal/timetable_termsofuse/"
-    let urlString: String = "https://yandex.ru/legal/practicum_offer"
+    let screenSize = UIScreen.main.bounds
+        //    let urlString: String = "https://yandex.ru/legal/timetable_termsofuse/"
```
Стоит в следующем спринте удалить все лишние комментарии

fury-sound committed:
sprint 20 - 1. Коллекция Сторис отображается, непросмотренные и просмотренные превью сторис отличаются визуально;
2.  Сверстан экран Сторис
3. По тапу на превью выбранная Сторис открывается на полный экран, можно просматривать изображения внутри сторис, они меняются по тапу или свайпу влево/вправо; можно перейти на предыдущую/следующую сторис по тапу/свайпу на первом изображении с краю слева, на последнем - с краю справа.
4. Экран Сторис закрывается по нажатию на кнопку в верхнем правом углу или по свайпу вниз.
5. Добавлена анимация открытия/закрытия экрана Сторис - спускается сверху/сворачивается в центре экрана

В тексте много остаточных комментариев.
